com.unity.mobile.notifications/Editor/AndroidNotificationPostProcessor.cs

```diff
@@ -1,303 +1,177 @@
+
+            manifest.Application.ReceiverList.AddElement(receiverNotificationRestartOnBoot);
+        }
+
+        private static void InjectAndroidManifest(Manifest manifest, ManifestSettings settings)
+        {
+            InjectReceivers(manifest);
+
+            if (settings.UseCustomActivity)
+                manifest.Application.AddMetaDataValue("custom_notification_android_activity", settings.CustomActivity);
+
+            if (settings.RescheduleOnRestart)
+            {
+                manifest.Application.AddMetaDataValue("reschedule_notifications_on_restart", "true");
+                manifest.AddUsesPermission("android.permission.RECEIVE_BOOT_COMPLETED");
+            }
+
+            manifest.AddUsesPermission("android.permission.POST_NOTIFICATIONS");
+
+            bool enableExact = (settings.ExactAlarm & AndroidExactSchedulingOption.ExactWhenAvailable) != 0;
+            manifest.Application.AddMetaDataValue("com.unity.androidnotifications.exact_scheduling", enableExact ? "1" : "0");
+            if (enableExact)
+            {
+                bool scheduleExact = (settings.ExactAlarm & AndroidExactSchedulingOption.AddScheduleExactPermission) != 0;
+                bool useExact = (settings.ExactAlarm & AndroidExactSchedulingOption.AddUseExactAlarmPermission) != 0;
+                // as documented here: https://developer.android.com/reference/android/Manifest.permission#USE_EXACT_ALARM
+                // only one of these two attributes should be used or max sdk set so on any device it's one or the other
+                if (scheduleExact)
+                {
+                    var sheduleExactAlarm = manifest.AddUsesPermission("android.permission.SCHEDULE_EXACT_ALARM");
+                    if (useExact)
+                        sheduleExactAlarm.Attributes.MaxSdkVersion.Set(32);
+                }
+                if (useExact)
+                    manifest.AddUsesPermission("android.permission.USE_EXACT_ALARM");
+
+                // Battery optimizations must use "uses-permission-sdk-23", regular uses-permission does not work
+                if ((settings.ExactAlarm & AndroidExactSchedulingOption.AddRequestIgnoreBatteryOptimizationsPermission) != 0)
+                {
+                    // TODO: Missing AddUsesPermissionSdk23 function
+                    var batterOptimizations = manifest.AddUsesPermission("android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS");
+                    batterOptimizations.Attributes.Name.Set("uses-permission-sdk-23");
+                }
+            } 
+        }
+
+        private static void InjectProguard(AndroidProjectFiles projectFiles)
+        {
+            // TODO: PropertyStringArray missing Add function?
+            var original = new List<string>(projectFiles.UnityLibraryBuildGradle.Android.DefaultConfig.ConsumerProguardFiles.Get());
+            original.Add(Path.GetFileName(NotificationProguardPath));

```

Comment:
@rytis-buz would it make sense to have projectFiles.UnityLibraryBuildGradle.Android.DefaultConfig.ConsumerProguardFiles.Add(string value) to append to array?

Reply:
Sure, it can be added